### PR TITLE
Move mixer check annotation to model with defaults

### PR DESCRIPTION
**Pull request by howardjohn**
Follow up refactoring for https://github.com/istio/istio/pull/11795

### CODE REVIEW

File: pilot/pkg/networking/plugin/mixer/mixer.go [hunk 1]
```diff
@@ -276,7 +271,8 @@ func buildTransport(mesh *meshconfig.MeshConfig, node *model.Proxy) *mccpb.Trans
 
 	networkFailPolicy := &mccpb.NetworkFailPolicy{Policy: policy}
 
-	if annotation, ok := node.Metadata[policyCheckRetriesAnnotation]; ok {
+	networkFailPolicy.MaxRetry = defaultRetries
```
Comment (kyessenov) [inline pilot/pkg/networking/plugin/mixer/mixer.go]: Can we _not_ set the config if it matches the hard-coded default in the proxy? This is proliferating config which hurts us in performance.

Comment (costinm) [reply pilot/pkg/networking/plugin/mixer/mixer.go]: +1 on avoiding config proliferation - unfortunately won't work. If we don't set it - proto3 defaults to 0, so envoy won't know if we want 0 or the built-in default.

File: pilot/pkg/networking/plugin/mixer/mixer.go [hunk 2]
```diff
@@ -285,7 +281,8 @@ func buildTransport(mesh *meshconfig.MeshConfig, node *model.Proxy) *mccpb.Trans
 		}
 	}
 
-	if annotation, ok := node.Metadata[policyCheckBaseRetryWaitTimeAnnotation]; ok {
+	networkFailPolicy.BaseRetryWait = types.DurationProto(defaultBaseRetryWaitTime)
```
Comment (kyessenov) [inline pilot/pkg/networking/plugin/mixer/mixer.go]: no need to create an object every time. make the object a constant.

File: pilot/pkg/networking/plugin/mixer/mixer.go [hunk 3]
```diff
@@ -294,7 +291,8 @@ func buildTransport(mesh *meshconfig.MeshConfig, node *model.Proxy) *mccpb.Trans
 		}
 	}
 
-	if annotation, ok := node.Metadata[policyCheckMaxRetryWaitTimeAnnotation]; ok {
+	networkFailPolicy.MaxRetryWait = types.DurationProto(defaultMaxRetryWaitTime)
```
Comment (kyessenov) [inline pilot/pkg/networking/plugin/mixer/mixer.go]: same here. make the default a duration proto.

**Comment by howardjohn** [comment]
@duderino can this be merged?

**Pull request merged**